amgut/lib/data_access/ag_data_access.py

```diff
@@ -969,3 +969,303 @@ def is_deposited_ebi(self, barcode):
             sql = "SELECT deposited FROM ag.ag_kit_barcodes WHERE barcode = %s"
             TRN.add(sql, [barcode])
             return TRN.execute_fetchlast()
+
+    # following are DB access functions only used for unit testing:
+
+    def get_random_supplied_kit_id_scanned_unconsented(self):
+        """ Returns randomly chosen supplied_kit_id and barcode which has been
+        scanned but is without consent.
+        For unit testing only!
+
+        Returns
+        -------
+        str: [supplied_kit_id, barcode]
+            example: ['fNIYa', '000001053']
+
+        Raises
+        ------
+        ValueError
+            If no kits can be found in the DB that have been scanned and
+            are without consent."""
+        with TRN:
+            sql = """SELECT supplied_kit_id, barcode
+                     FROM barcodes.barcode
+                     JOIN ag.ag_kit_barcodes USING (barcode)
+                     JOIN ag.ag_kit USING (ag_kit_id)
+                     WHERE barcodes.barcode.scan_date IS NOT NULL
+                     AND ag.ag_kit_barcodes.survey_id IS NULL
+                     LIMIT 1"""
+            TRN.add(sql, [])
+            info = TRN.execute_fetchindex()
+            if not info:
+                raise ValueError('No kits found.')
+            return info[0]
+
+    def get_random_handout_printed_min6_supplied_kit_id(self):
+        """ Returns a random supplied_kit_id with printed results
+        and 6 swaps per kit.
+        For unit testing only!
+
+        Returns
+        -------
+        supplied_kit_id : str
+            A supplied_kit_id. Example: 'DS_ubdvq'
+
+        Raises
+        ------
+        ValueError
+            If no hand out kit exists, satisfing the given conditions."""
+        with TRN:
+            sql = """SELECT kit_id
+                     FROM ag.ag_handout_kits
+                     WHERE swabs_per_kit = 6 AND print_results = TRUE"""
+            TRN.add(sql, [])
+            info = TRN.execute_fetchindex()
+            if not info:
+                raise ValueError('No kits found.')
+            return info[0][0]
+
+    def get_random_email(self):
+        """ Return randomly chosen email.
+        For unit testing only!
+
+        Returns
+        -------
+        str: email
+            Example: 'a03E9u6ZAu@glA+)./Vn'
+
+        Raises
+        ------
+        ValueError
+            If no emails be found in the DB."""
+        with TRN:
+            sql = """SELECT email
+                     FROM ag.ag_login
+                     LIMIT 1"""
+            TRN.add(sql, [])
+            info = TRN.execute_fetchindex()
+            if not info:
+                raise ValueError('No emails found.')
+            return info[0][0]
+
+    def get_random_barcode(self, deposited=True):
+        """ Returns randomly chosen barcode.
```

Comment:
maybe random is not the right word. It returns the first hit in the DB (by using limit 1 in the SQL statement), thus it depends on the internal indexing of the DB which value is returned. Let's say it is quasi random ;-)